### PR TITLE
Use hyphens over underscores when generating icav2 paths

### LINKED ISSUE #349: Use hy-phens over under_scores in project data naming

### PR DESCRIPTION
**Pull request by alexiswl**
Resolves #349 

Also Fixed shell script to point to development, instead of trial project in some examples

### CODE REVIEW

File: config/constants.ts [hunk 1]
```diff
@@ -200,8 +200,8 @@ export const mockInstrumentRunTableName = 'stacky-instrument-run-table';
...
 export const mockWorkflowManagerTableName = 'stacky-workflow-manager-table';
 export const mockIcav2ProjectIdSsmParameterName = '/orcabus/stacky/icav2_project_id_and_name_json';
-export const mockPrimaryOutputUriSsmParameterName = '/orcabus/stacky/primary_output_uri'; // icav2://7595e8f2-32d3-4c76-a324-c6a85dae87b5/primary_data/__instrument_run_id__/__portal_run_id__/
-export const mockAnalysisOutputUriSsmParameterName = '/orcabus/stacky/analysis_output_uri'; // icav2://7595e8f2-32d3-4c76-a324-c6a85dae87b5/analysis_data/__workflow_name__/__workflow_version__/__portal_run_id__/
-export const mockAnalysisLogsUriSsmParameterName = '/orcabus/stacky/analysis_logs_uri'; // icav2://7595e8f2-32d3-4c76-a324-c6a85dae87b5/analysis_logs/__workflow_name__/__workflow_version__/__portal_run_id__/
+export const mockPrimaryOutputUriSsmParameterName = '/orcabus/stacky/primary_output_uri'; // icav2://7595e8f2-32d3-4c76-a324-c6a85dae87b5/primary/__instrument_run_id__/__portal_run_id__/
+export const mockAnalysisOutputUriSsmParameterName = '/orcabus/stacky/analysis_output_uri'; // icav2://7595e8f2-32d3-4c76-a324-c6a85dae87b5/analysis/__workflow_name__/__portal_run_id__/
```
Comment (victorskl) [inline config/constants.ts:204]: Just confirming; do we want to remove `-data` suffix from the mid path too?

No strong preference from me. If we decided so, I will update [the design doc](https://umccr.slack.com/archives/C025TLC7D/p1718269513888569?thread_ts=1717638564.248829&cid=C025TLC7D) accordingly.

@pdiakumis @reisingerf 

This must be the last bit that I might have missed yesterday discussion, Sorry. Let us finalise here, folks.

File: lib/workload/stateless/stacks/bclconvert-interop-qc-pipeline-manager/scripts/trigger_bclconvert_interop_qc.sh [hunk 2]
```diff
@@ -21,19 +21,19 @@ WORKFLOW_VERSION="1.3.1--1.21__20240410040204"
...
+PROJECT_NAME="development"
+PROJECT_ID="ea19a3f5-ec7c-4940-a474-c31cd91dbad4"
+WORKFLOW_RUN_NAME_PREFIX="umccr__automated__${WORKFLOW_NAME}__1_3_1__1_21__20240410040204__"
 PORTAL_RUN_ID="$(date +%Y%m%d)$(xxd -l 4 -c 4 -p < /dev/random)"
 
 # Outputs
-ANALYSIS_OUTPUT_URI_PREFIX="icav2://7595e8f2-32d3-4c76-a324-c6a85dae87b5/analysis_data/bclconvert_interop_qc/1_3_1--1_21__20240410040204/"
-ICA_LOGS_URI_PREFIX="icav2://7595e8f2-32d3-4c76-a324-c6a85dae87b5/analysis_logs/bclconvert_interop_qc/1_3_1--1_21__20240410040204/"
+ANALYSIS_OUTPUT_URI_PREFIX="icav2://${PROJECT_ID}/analysis_data/${WORKFLOW_NAME}/1_3_1__1_21__20240410040204/"
+ICA_LOGS_URI_PREFIX="icav2://${PROJECT_ID}/analysis_logs/${WORKFLOW_NAME}/${WORKFLOW_VERSION//./_}/"
```
Comment (victorskl) [inline lib/workload/stateless/stacks/bclconvert-interop-qc-pipeline-manager/scripts/trigger_bclconvert_interop_qc.sh:36]: Perhaps, these mid paths might have to change accordingly too, Alexis? 
Lines 24, 25, 35, 36.

File: lib/workload/stateless/stacks/bssh-icav2-fastq-copy-manager/lambdas/query_bclconvert_outputs_handler_py/query_bclconvert_outputs_handler.py [hunk 3]
```diff
@@ -302,22 +302,22 @@ def handler(event, context):
...
-    # }
+# if __name__ == "__main__":
+#     print(
+#         json.dumps(
+#             handler(
+#                 {
+#                     "project_id": "a7c67a80-c8f2-4348-adec-3a5a073d1d55",
+#                     "analysis_id": "2800e302-4a12-4bf0-9028-f74e4a83479f",
+#                     "output_uri": "icav2://ea19a3f5-ec7c-4940-a474-c31cd91dbad4/primary_data/240424_A01052_0193_BH7JMMDRX5/202406066dd81cc9/"
+#                 },
```
Comment (victorskl) [inline lib/workload/stateless/stacks/bssh-icav2-fastq-copy-manager/lambdas/query_bclconvert_outputs_handler_py/query_bclconvert_outputs_handler.py:313]: Same, mid path.

File: config/constants.ts (hunk 1 above)
Comment (reisingerf) [reply config/constants.ts:204]: No strong feelings either, but I'd personally would keep the "-data". 
It's more specifically talking about "data" and is less easily confused. For example AWS has primary (and secondary) analysis workflows, the outputs of which would NOT fall under our "primary" definition.

Comment (pdiakumis) [reply config/constants.ts:204]: Not fussed either. Compare:

- `analysis-data`
- `analysis-cache`
- `primary-data`

vs.

- `analysis`
- `analysis-cache`
- `primary`

File: lib/workload/stateless/stacks/bclconvert-interop-qc-pipeline-manager/scripts/trigger_bclconvert_interop_qc.sh [hunk 4]
```diff
@@ -21,19 +21,19 @@ WORKFLOW_VERSION="1.3.1--1.21__20240410040204"
 #############
 
 # Inputs
-BCLCONVERT_REPORT_DIRECTORY_URI="icav2://7595e8f2-32d3-4c76-a324-c6a85dae87b5/primary_data/240229_A00130_0288_BH5HM2DSXC/202405302b817f5e/Reports/"
-INTEROP_DIRECTORY_URI="icav2://7595e8f2-32d3-4c76-a324-c6a85dae87b5/primary_data/240229_A00130_0288_BH5HM2DSXC/202405302b817f5e/InterOp/"
-INSTRUMENT_RUN_ID="240229_A00130_0288_BH5HM2DSXC"
+BCLCONVERT_REPORT_DIRECTORY_URI="icav2://ea19a3f5-ec7c-4940-a474-c31cd91dbad4/primary_data/240424_A01052_0193_BH7JMMDRX5/20240606b794561d/Reports/"
```
Comment (reisingerf) [inline lib/workload/stateless/stacks/bclconvert-interop-qc-pipeline-manager/scripts/trigger_bclconvert_interop_qc.sh:24]: Out of curiosity: Why are some "Icav2://" path with project ID and others with project name?

File: config/constants.ts (hunk 1 above)
Comment (victorskl) [reply config/constants.ts:204]: Okay. @alexiswl any preference about `-data` suffix?

Comment (victorskl) [reply config/constants.ts:204]: I've gathered the feedback. Team members' view are flexible on either way. So, making a call here as tiebreaker.

Let us go without `-data` suffix for this round of iteration for the mid-path. So, just `/primary/` and `/analysis/`. I will update the orcabus design doc.

In this case, no change is needed here anymore.

**Review by victorskl — APPROVED** [review]
LGTM from my side.

I will need to merge before starting rounds of testing, next. So, it would be great we can merge it soon.

**Comment by victorskl** [comment]
Updated the doc. It is official now. Yay.!

The final output structure will like:

```
s3://<bucket-name>/byob-icav2/<project-name>/<analysis|primary>/<workflow-name>/<portal-run-id>/...
```

**Comment by victorskl** [comment]
@reisingerf @pdiakumis Flo/Pete; if this is ok, could you help approve PR, pls.

**Review by pdiakumis — APPROVED** [review]

File: lib/workload/stateless/stacks/bclconvert-interop-qc-pipeline-manager/scripts/trigger_bclconvert_interop_qc.sh (hunk 4 above)
Comment (alexiswl) [reply lib/workload/stateless/stacks/bclconvert-interop-qc-pipeline-manager/scripts/trigger_bclconvert_interop_qc.sh:24]: It doesn't matter, if it's not a uuid, wrapica converts from a project name to a project id. Mostly using project ids

**Review by reisingerf — APPROVED** [review]

**Comment by victorskl** [comment]
@alexiswl Could you merge it when you are ready, pls.

**Pull request merged**